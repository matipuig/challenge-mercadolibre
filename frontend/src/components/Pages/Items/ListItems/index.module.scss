.container {
  background-color: #fff;
  display: flex;
  flex-direction: column;
  gap: 5px;
}
