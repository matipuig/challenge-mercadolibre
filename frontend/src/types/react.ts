/**
 * Contains the different types necessary to use React with typescript.
 */
import React, { PropsWithChildren, ReactElement as Element } from 'react';

export type ComponentChildren = React.ReactNode;
export type ReactElement = Element;
export type FunctionalComponent<P = unknown> = (props: P) => ReactElement | null;
export type FunctionalComponentWithChildren<P = unknown> = FunctionalComponent<
  PropsWithChildren<P>
>;
