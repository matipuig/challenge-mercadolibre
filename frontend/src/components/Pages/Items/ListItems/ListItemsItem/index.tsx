/**
 * Contains the list items item
 */
import { ReactElement } from 'react';

import Image from 'next/image';
import Link from 'next/link';
import { useTranslation } from 'react-i18next';

import { PriceDisplay } from '~/components/Pages/Common/PriceDisplay';
import { CONSTANTS } from '~/constants';
import { getAvailableI18nTexts, i18nReplace } from '~/i18n';
import { Item, SellerAddress } from '~/types/services/backend';
import { getURLForPublicContent } from '~/utils/components';
import { getURLFriendlyString } from '~/utils/queryParams';

import styles from './index.module.scss';

interface ListItemsItemProps {
  item: Item;
}

const { ROUTES } = CONSTANTS;
const { QUERY_PARAMS } = ROUTES;
const texts = getAvailableI18nTexts();
const { productImageAlt, freeShippingImage } = texts.components.listItems;

const getAddress = (sellerAddress: SellerAddress): string => {
  const { city, state } = sellerAddress;
  return `${state.name}, ${city.name}`;
};

const getIitemLinkDirection = (listItem: Item): string => {
  const queriedTitle = getURLFriendlyString(listItem.title);
  const urlWithId = ROUTES.ITEM.replace(':id', listItem.id);
  return `${urlWithId}?${QUERY_PARAMS.DESCRIPTION}=${queriedTitle}`;
};

export const ListItemsItem = ({ item }: ListItemsItemProps): ReactElement => {
  const { t } = useTranslation();
  return (
    <article key={item.id} className={styles.listItem} data-testid="listItemsItem">
      <figure className={styles.imageContainer} data-testid="listItemsItemImage">
        <Link href={getIitemLinkDirection(item)}>
          <a>
            <Image
              className={styles.image}
              src={item.picture}
              width={160}
              height={160}
              quality={100}
              alt={i18nReplace(t(productImageAlt), item.title)}
            />
          </a>
        </Link>
      </figure>
      <div className={styles.descriptionContainer}>
        <div className={styles.descriptionUpperPart}>
          <div className={styles.priceAndShippingContainer}>
            <div className={styles.price}>
              <PriceDisplay price={item.price} withDecimals={false} />
            </div>
            <div className={styles.freeShipping}>
              {item.free_shipping || (
                <Image
                  src={getURLForPublicContent('/images/icons/shipping-small.png')}
                  alt={i18nReplace(t(freeShippingImage.alt), item.title)}
                  height={16}
                  title={t(freeShippingImage.title)}
                  width={16}
                  quality={100}
                />
              )}
            </div>
          </div>
          <div className={styles.address}>{getAddress(item.seller_address)}</div>
        </div>
        <Link href={getIitemLinkDirection(item)}>
          <a>
            <h2 className={styles.title}>{item.title}</h2>
          </a>
        </Link>
      </div>
    </article>
  );
};

export default ListItemsItem;
