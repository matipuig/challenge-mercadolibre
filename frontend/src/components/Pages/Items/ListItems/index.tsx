/**
 * Contains the no results screen.
 */
import { ReactElement } from 'react';

import Image from 'next/image';
import Link from 'next/link';
import { useTranslation } from 'react-i18next';

import { CONSTANTS } from '~/constants';
import { getAvailableI18nTexts } from '~/i18n';
import { Item, Price, SellerAddress } from '~/types/services/backend';
import { makeHumanFriendly } from '~/utils/conversors';

import styles from './index.module.scss';

interface ListItemsProps {
  items: Item[];
}

const { ROUTES } = CONSTANTS;
const texts = getAvailableI18nTexts();
const { productImageAlt, freeShippingImage } = texts.components.listItems;

const getPrice = (price: Price): string => {
  const { currency, amount } = price;
  const currencySign = currency !== 'ARS' ? currency : '$';
  const humanFriendlyAmount = makeHumanFriendly(amount);
  return `${currencySign} ${humanFriendlyAmount}`;
};

const getAddress = (sellerAddress: SellerAddress): string => {
  const { city, state } = sellerAddress;
  return `${state.name}, ${city.name}`;
};

const getIitemLinkDirection = (listItem: Item): string => ROUTES.ITEM.replace(':id', listItem.id);

export const ListItems = ({ items }: ListItemsProps): ReactElement => {
  const { t } = useTranslation();
  return (
    <section className={styles.container}>
      {items.map((item) => (
        <article key={item.id} className={styles.listItem}>
          <figure className={styles.imageContainer}>
            <Link href={getIitemLinkDirection(item)}>
              <a>
                <Image
                  className={styles.image}
                  src={item.picture}
                  width={160}
                  height={160}
                  quality="100"
                  alt={t(productImageAlt).replace('[?]', item.title)}
                />
              </a>
            </Link>
          </figure>
          <div className={styles.descriptionContainer}>
            <div className={styles.descriptionUpperPart}>
              <div className={styles.priceAndShippingContainer}>
                <div className={styles.price}>{getPrice(item.price)}</div>
                <div className={styles.freeShipping}>
                  {item.free_shipping || (
                    <Image
                      src="/images/icons/shipping-small.png"
                      alt={t(freeShippingImage.alt).replace('[?]', item.title)}
                      title={t(freeShippingImage.title)}
                      width={16}
                      height={16}
                      quality={100}
                    />
                  )}
                </div>
              </div>
              <div className={styles.address}>{getAddress(item.seller_address)}</div>
            </div>
            <Link href={getIitemLinkDirection(item)}>
              <a>
                <h2 className={styles.title}>{item.title}</h2>
              </a>
            </Link>
          </div>
        </article>
      ))}
    </section>
  );
};

export default ListItems;
