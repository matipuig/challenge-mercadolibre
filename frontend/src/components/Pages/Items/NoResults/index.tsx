/**
 * Contains the no results solution.
 */
import { ReactElement } from 'react';

import Image from 'next/image';
import { useTranslation } from 'react-i18next';

import { getAvailableI18nTexts } from '~/i18n';
import { getURLForPublicContent } from '~/utils/components';

import styles from './index.module.scss';

const texts = getAvailableI18nTexts();
const { mainText, textAdvice } = texts.components.listItems.noResults;

export const NoResults = (): ReactElement => {
  const { t } = useTranslation();
  return (
    <section className={styles.container} data-testid="noResultsComponent">
      <div className={styles.imageContainer}>
        <Image
          className={styles.image}
          src={getURLForPublicContent('/images/pictures/noResults.svg')}
          width={250}
          height={100}
          alt=""
          quality={100}
        />
      </div>
      <h3 className={styles.mainText}>{t(mainText)}</h3>
      <p className={styles.textAdvice}>{t(textAdvice)}</p>
    </section>
  );
};
