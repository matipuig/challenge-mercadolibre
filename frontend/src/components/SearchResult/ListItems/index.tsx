/**
 * Contains the no results screen.
 */
import { ReactElement } from 'react';
import Image from 'next/image';

import styles from './index.module.scss';
import { Item, Price, SellerAddress } from '~/types/services/backend';

interface ListItemsProps {
  items: Item[];
}

const getPrice = (price: Price): string => {
  const { currency, amount } = price;
  const currencySign = currency !== 'ARS' ? currency : '$';
  return `${currencySign} ${amount}`;
};

const getAddress = (sellerAddress: SellerAddress): string => {
  const { city, state } = sellerAddress;
  return `${state.name}, ${city.name}`;
};

export const ListItems = ({ items }: ListItemsProps): ReactElement => {
  return (
    <div className={styles.container}>
      {items.map((item) => (
        <div className={styles.listItem}>
          <div className={styles.imageContainer}>
            <Image className={styles.image} src={item.picture} width={180} height={180} />
          </div>
          <div className={styles.descriptionContainer}>
            <div className={styles.descriptionUpperPart}>
              <div className={styles.priceAndShippingContainer}>
                <div className={styles.price}>{getPrice(item.price)}</div>
                <div className={styles.freeShipping}>
                  {item.free_shipping || <img src={'/icons/shipping-small.png'} />}
                </div>
              </div>
              <div className={styles.place}>{getAddress(item.seller_address)}</div>
            </div>
            <div className={styles.title}>{item.title}</div>
          </div>
        </div>
      ))}
    </div>
  );
};

export default ListItems;
