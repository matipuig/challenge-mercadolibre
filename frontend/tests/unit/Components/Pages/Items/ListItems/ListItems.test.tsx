/**
 * Contains the unit tests for the list items.
 */
import { cleanup, render, RenderResult } from '@testing-library/react';
import '@testing-library/jest-dom/extend-expect';

import { ListItems } from '../../../../../../src/components/Pages/Items/ListItems';
import { mockedListItems } from '../../../../__mocks__/listItems';

describe('List items component', () => {
  let listItems: RenderResult;

  beforeEach(() => {
    listItems = render(<ListItems items={mockedListItems} />);
  });

  afterEach(() => {
    cleanup();
  });

  it('renders.', () => {
    expect(listItems.baseElement).toBeInTheDocument();
  });

  it('matches snapshot.', () => {
    expect(listItems).toMatchSnapshot();
  });

  it('has the section tag.', () => {
    const sectionTag = listItems.baseElement.querySelector('section');
    expect(sectionTag).toBeDefined();
  });

  it('has all the list items.', () => {
    const innerListItems = listItems.getAllByTestId('listItemsItem');
    expect(innerListItems.length).toEqual(mockedListItems.length);
  });
});
