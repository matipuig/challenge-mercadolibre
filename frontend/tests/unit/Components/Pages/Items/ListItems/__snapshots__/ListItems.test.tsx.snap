// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`List items component matches snapshot. 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <section>
        <article
          data-testid="listItemsItem"
        >
          <figure
            data-testid="listItemsItemImage"
          >
            <a
              href="/items/MLA1144829687?description=Motosierra+A+Nafta+Philco+52cc+Mjpmo217+2.4hp"
            >
              <span
                style="box-sizing: border-box; display: inline-block; overflow: hidden; background: none; opacity: 1; border: 0px; margin: 0px; padding: 0px; position: relative; max-width: 100%;"
              >
                <span
                  style="box-sizing: border-box; display: block; background: none; opacity: 1; border: 0px; margin: 0px; padding: 0px; max-width: 100%;"
                >
                  <img
                    alt=""
                    aria-hidden="true"
                    src="data:image/svg+xml,%3csvg%20xmlns=%27http://www.w3.org/2000/svg%27%20version=%271.1%27%20width=%27160%27%20height=%27160%27/%3e"
                    style="display: block; max-width: 100%; background: none; opacity: 1; border: 0px; margin: 0px; padding: 0px;"
                  />
                </span>
                <img
                  alt="Imagen de Motosierra A Nafta Philco 52cc Mjpmo217 2.4hp."
                  data-nimg="intrinsic"
                  decoding="async"
                  src="data:image/gif;base64,R0lGODlhAQABAIAAAAAAAP///yH5BAEAAAAALAAAAAABAAEAAAIBRAA7"
                  style="position: absolute; top: 0px; left: 0px; bottom: 0px; right: 0px; box-sizing: border-box; padding: 0px; margin: auto; display: block; width: 0px; height: 0px; min-width: 100%; max-width: 100%; min-height: 100%; max-height: 100%;"
                />
                <noscript />
              </span>
            </a>
          </figure>
          <div>
            <div>
              <div>
                <div>
                  <div
                    aria-label="17399 pesos"
                    data-testid="priceDisplay"
                  >
                    <span>
                      $
                       
                      17.399
                    </span>
                  </div>
                </div>
                <div />
              </div>
              <div>
                Córdoba, Córdoba
              </div>
            </div>
            <a
              href="/items/MLA1144829687?description=Motosierra+A+Nafta+Philco+52cc+Mjpmo217+2.4hp"
            >
              <h2>
                Motosierra A Nafta Philco 52cc Mjpmo217 2.4hp
              </h2>
            </a>
          </div>
        </article>
        <article
          data-testid="listItemsItem"
        >
          <figure
            data-testid="listItemsItemImage"
          >
            <a
              href="/items/MLA926186472?description=Silla+Plastica+X+10+Apilable+Reforzada+Negro+Reina+Quality+"
            >
              <span
                style="box-sizing: border-box; display: inline-block; overflow: hidden; background: none; opacity: 1; border: 0px; margin: 0px; padding: 0px; position: relative; max-width: 100%;"
              >
                <span
                  style="box-sizing: border-box; display: block; background: none; opacity: 1; border: 0px; margin: 0px; padding: 0px; max-width: 100%;"
                >
                  <img
                    alt=""
                    aria-hidden="true"
                    src="data:image/svg+xml,%3csvg%20xmlns=%27http://www.w3.org/2000/svg%27%20version=%271.1%27%20width=%27160%27%20height=%27160%27/%3e"
                    style="display: block; max-width: 100%; background: none; opacity: 1; border: 0px; margin: 0px; padding: 0px;"
                  />
                </span>
                <img
                  alt="Imagen de Silla Plástica X 10 Apilable Reforzada Negro Reina Quality ."
                  data-nimg="intrinsic"
                  decoding="async"
                  src="data:image/gif;base64,R0lGODlhAQABAIAAAAAAAP///yH5BAEAAAAALAAAAAABAAEAAAIBRAA7"
                  style="position: absolute; top: 0px; left: 0px; bottom: 0px; right: 0px; box-sizing: border-box; padding: 0px; margin: auto; display: block; width: 0px; height: 0px; min-width: 100%; max-width: 100%; min-height: 100%; max-height: 100%;"
                />
                <noscript />
              </span>
            </a>
          </figure>
          <div>
            <div>
              <div>
                <div>
                  <div
                    aria-label="19199 pesos"
                    data-testid="priceDisplay"
                  >
                    <span>
                      $
                       
                      19.199
                    </span>
                  </div>
                </div>
                <div />
              </div>
              <div>
                Santa Fe, Villa Gobernador Gálvez
              </div>
            </div>
            <a
              href="/items/MLA926186472?description=Silla+Plastica+X+10+Apilable+Reforzada+Negro+Reina+Quality+"
            >
              <h2>
                Silla Plástica X 10 Apilable Reforzada Negro Reina Quality 
              </h2>
            </a>
          </div>
        </article>
        <article
          data-testid="listItemsItem"
        >
          <figure
            data-testid="listItemsItemImage"
          >
            <a
              href="/items/MLA850868389?description=Sopladora+Aspiradora+Philco+Mjpsa116+Electrica+2600w+220v+-+240v"
            >
              <span
                style="box-sizing: border-box; display: inline-block; overflow: hidden; background: none; opacity: 1; border: 0px; margin: 0px; padding: 0px; position: relative; max-width: 100%;"
              >
                <span
                  style="box-sizing: border-box; display: block; background: none; opacity: 1; border: 0px; margin: 0px; padding: 0px; max-width: 100%;"
                >
                  <img
                    alt=""
                    aria-hidden="true"
                    src="data:image/svg+xml,%3csvg%20xmlns=%27http://www.w3.org/2000/svg%27%20version=%271.1%27%20width=%27160%27%20height=%27160%27/%3e"
                    style="display: block; max-width: 100%; background: none; opacity: 1; border: 0px; margin: 0px; padding: 0px;"
                  />
                </span>
                <img
                  alt="Imagen de Sopladora Aspiradora Philco Mjpsa116  Eléctrica 2600w 220v - 240v."
                  data-nimg="intrinsic"
                  decoding="async"
                  src="data:image/gif;base64,R0lGODlhAQABAIAAAAAAAP///yH5BAEAAAAALAAAAAABAAEAAAIBRAA7"
                  style="position: absolute; top: 0px; left: 0px; bottom: 0px; right: 0px; box-sizing: border-box; padding: 0px; margin: auto; display: block; width: 0px; height: 0px; min-width: 100%; max-width: 100%; min-height: 100%; max-height: 100%;"
                />
                <noscript />
              </span>
            </a>
          </figure>
          <div>
            <div>
              <div>
                <div>
                  <div
                    aria-label="10899 pesos"
                    data-testid="priceDisplay"
                  >
                    <span>
                      $
                       
                      10.899
                    </span>
                  </div>
                </div>
                <div />
              </div>
              <div>
                Buenos Aires, Tristán Suarez
              </div>
            </div>
            <a
              href="/items/MLA850868389?description=Sopladora+Aspiradora+Philco+Mjpsa116+Electrica+2600w+220v+-+240v"
            >
              <h2>
                Sopladora Aspiradora Philco Mjpsa116  Eléctrica 2600w 220v - 240v
              </h2>
            </a>
          </div>
        </article>
        <article
          data-testid="listItemsItem"
        >
          <figure
            data-testid="listItemsItemImage"
          >
            <a
              href="/items/MLA1121494420?description=Sopladora+Daewoo+Daeb600+Electrica+600w+240v"
            >
              <span
                style="box-sizing: border-box; display: inline-block; overflow: hidden; background: none; opacity: 1; border: 0px; margin: 0px; padding: 0px; position: relative; max-width: 100%;"
              >
                <span
                  style="box-sizing: border-box; display: block; background: none; opacity: 1; border: 0px; margin: 0px; padding: 0px; max-width: 100%;"
                >
                  <img
                    alt=""
                    aria-hidden="true"
                    src="data:image/svg+xml,%3csvg%20xmlns=%27http://www.w3.org/2000/svg%27%20version=%271.1%27%20width=%27160%27%20height=%27160%27/%3e"
                    style="display: block; max-width: 100%; background: none; opacity: 1; border: 0px; margin: 0px; padding: 0px;"
                  />
                </span>
                <img
                  alt="Imagen de Sopladora Daewoo Daeb600  Eléctrica 600w 240v."
                  data-nimg="intrinsic"
                  decoding="async"
                  src="data:image/gif;base64,R0lGODlhAQABAIAAAAAAAP///yH5BAEAAAAALAAAAAABAAEAAAIBRAA7"
                  style="position: absolute; top: 0px; left: 0px; bottom: 0px; right: 0px; box-sizing: border-box; padding: 0px; margin: auto; display: block; width: 0px; height: 0px; min-width: 100%; max-width: 100%; min-height: 100%; max-height: 100%;"
                />
                <noscript />
              </span>
            </a>
          </figure>
          <div>
            <div>
              <div>
                <div>
                  <div
                    aria-label="7820 pesos"
                    data-testid="priceDisplay"
                  >
                    <span>
                      $
                       
                      7.820
                    </span>
                  </div>
                </div>
                <div />
              </div>
              <div>
                Buenos Aires, San Justo
              </div>
            </div>
            <a
              href="/items/MLA1121494420?description=Sopladora+Daewoo+Daeb600+Electrica+600w+240v"
            >
              <h2>
                Sopladora Daewoo Daeb600  Eléctrica 600w 240v
              </h2>
            </a>
          </div>
        </article>
        <article
          data-testid="listItemsItem"
        >
          <figure
            data-testid="listItemsItemImage"
          >
            <a
              href="/items/MLA883353487?description=Sillas+Plasticas+Reforzadas+Apilables+Reina+Pack+10"
            >
              <span
                style="box-sizing: border-box; display: inline-block; overflow: hidden; background: none; opacity: 1; border: 0px; margin: 0px; padding: 0px; position: relative; max-width: 100%;"
              >
                <span
                  style="box-sizing: border-box; display: block; background: none; opacity: 1; border: 0px; margin: 0px; padding: 0px; max-width: 100%;"
                >
                  <img
                    alt=""
                    aria-hidden="true"
                    src="data:image/svg+xml,%3csvg%20xmlns=%27http://www.w3.org/2000/svg%27%20version=%271.1%27%20width=%27160%27%20height=%27160%27/%3e"
                    style="display: block; max-width: 100%; background: none; opacity: 1; border: 0px; margin: 0px; padding: 0px;"
                  />
                </span>
                <img
                  alt="Imagen de Sillas Plasticas Reforzadas Apilables Reina Pack 10."
                  data-nimg="intrinsic"
                  decoding="async"
                  src="data:image/gif;base64,R0lGODlhAQABAIAAAAAAAP///yH5BAEAAAAALAAAAAABAAEAAAIBRAA7"
                  style="position: absolute; top: 0px; left: 0px; bottom: 0px; right: 0px; box-sizing: border-box; padding: 0px; margin: auto; display: block; width: 0px; height: 0px; min-width: 100%; max-width: 100%; min-height: 100%; max-height: 100%;"
                />
                <noscript />
              </span>
            </a>
          </figure>
          <div>
            <div>
              <div>
                <div>
                  <div
                    aria-label="20490 pesos"
                    data-testid="priceDisplay"
                  >
                    <span>
                      $
                       
                      20.490
                    </span>
                  </div>
                </div>
                <div />
              </div>
              <div>
                Buenos Aires, Vicente López
              </div>
            </div>
            <a
              href="/items/MLA883353487?description=Sillas+Plasticas+Reforzadas+Apilables+Reina+Pack+10"
            >
              <h2>
                Sillas Plasticas Reforzadas Apilables Reina Pack 10
              </h2>
            </a>
          </div>
        </article>
      </section>
    </div>
  </body>,
  "container": <div>
    <section>
      <article
        data-testid="listItemsItem"
      >
        <figure
          data-testid="listItemsItemImage"
        >
          <a
            href="/items/MLA1144829687?description=Motosierra+A+Nafta+Philco+52cc+Mjpmo217+2.4hp"
          >
            <span
              style="box-sizing: border-box; display: inline-block; overflow: hidden; background: none; opacity: 1; border: 0px; margin: 0px; padding: 0px; position: relative; max-width: 100%;"
            >
              <span
                style="box-sizing: border-box; display: block; background: none; opacity: 1; border: 0px; margin: 0px; padding: 0px; max-width: 100%;"
              >
                <img
                  alt=""
                  aria-hidden="true"
                  src="data:image/svg+xml,%3csvg%20xmlns=%27http://www.w3.org/2000/svg%27%20version=%271.1%27%20width=%27160%27%20height=%27160%27/%3e"
                  style="display: block; max-width: 100%; background: none; opacity: 1; border: 0px; margin: 0px; padding: 0px;"
                />
              </span>
              <img
                alt="Imagen de Motosierra A Nafta Philco 52cc Mjpmo217 2.4hp."
                data-nimg="intrinsic"
                decoding="async"
                src="data:image/gif;base64,R0lGODlhAQABAIAAAAAAAP///yH5BAEAAAAALAAAAAABAAEAAAIBRAA7"
                style="position: absolute; top: 0px; left: 0px; bottom: 0px; right: 0px; box-sizing: border-box; padding: 0px; margin: auto; display: block; width: 0px; height: 0px; min-width: 100%; max-width: 100%; min-height: 100%; max-height: 100%;"
              />
              <noscript />
            </span>
          </a>
        </figure>
        <div>
          <div>
            <div>
              <div>
                <div
                  aria-label="17399 pesos"
                  data-testid="priceDisplay"
                >
                  <span>
                    $
                     
                    17.399
                  </span>
                </div>
              </div>
              <div />
            </div>
            <div>
              Córdoba, Córdoba
            </div>
          </div>
          <a
            href="/items/MLA1144829687?description=Motosierra+A+Nafta+Philco+52cc+Mjpmo217+2.4hp"
          >
            <h2>
              Motosierra A Nafta Philco 52cc Mjpmo217 2.4hp
            </h2>
          </a>
        </div>
      </article>
      <article
        data-testid="listItemsItem"
      >
        <figure
          data-testid="listItemsItemImage"
        >
          <a
            href="/items/MLA926186472?description=Silla+Plastica+X+10+Apilable+Reforzada+Negro+Reina+Quality+"
          >
            <span
              style="box-sizing: border-box; display: inline-block; overflow: hidden; background: none; opacity: 1; border: 0px; margin: 0px; padding: 0px; position: relative; max-width: 100%;"
            >
              <span
                style="box-sizing: border-box; display: block; background: none; opacity: 1; border: 0px; margin: 0px; padding: 0px; max-width: 100%;"
              >
                <img
                  alt=""
                  aria-hidden="true"
                  src="data:image/svg+xml,%3csvg%20xmlns=%27http://www.w3.org/2000/svg%27%20version=%271.1%27%20width=%27160%27%20height=%27160%27/%3e"
                  style="display: block; max-width: 100%; background: none; opacity: 1; border: 0px; margin: 0px; padding: 0px;"
                />
              </span>
              <img
                alt="Imagen de Silla Plástica X 10 Apilable Reforzada Negro Reina Quality ."
                data-nimg="intrinsic"
                decoding="async"
                src="data:image/gif;base64,R0lGODlhAQABAIAAAAAAAP///yH5BAEAAAAALAAAAAABAAEAAAIBRAA7"
                style="position: absolute; top: 0px; left: 0px; bottom: 0px; right: 0px; box-sizing: border-box; padding: 0px; margin: auto; display: block; width: 0px; height: 0px; min-width: 100%; max-width: 100%; min-height: 100%; max-height: 100%;"
              />
              <noscript />
            </span>
          </a>
        </figure>
        <div>
          <div>
            <div>
              <div>
                <div
                  aria-label="19199 pesos"
                  data-testid="priceDisplay"
                >
                  <span>
                    $
                     
                    19.199
                  </span>
                </div>
              </div>
              <div />
            </div>
            <div>
              Santa Fe, Villa Gobernador Gálvez
            </div>
          </div>
          <a
            href="/items/MLA926186472?description=Silla+Plastica+X+10+Apilable+Reforzada+Negro+Reina+Quality+"
          >
            <h2>
              Silla Plástica X 10 Apilable Reforzada Negro Reina Quality 
            </h2>
          </a>
        </div>
      </article>
      <article
        data-testid="listItemsItem"
      >
        <figure
          data-testid="listItemsItemImage"
        >
          <a
            href="/items/MLA850868389?description=Sopladora+Aspiradora+Philco+Mjpsa116+Electrica+2600w+220v+-+240v"
          >
            <span
              style="box-sizing: border-box; display: inline-block; overflow: hidden; background: none; opacity: 1; border: 0px; margin: 0px; padding: 0px; position: relative; max-width: 100%;"
            >
              <span
                style="box-sizing: border-box; display: block; background: none; opacity: 1; border: 0px; margin: 0px; padding: 0px; max-width: 100%;"
              >
                <img
                  alt=""
                  aria-hidden="true"
                  src="data:image/svg+xml,%3csvg%20xmlns=%27http://www.w3.org/2000/svg%27%20version=%271.1%27%20width=%27160%27%20height=%27160%27/%3e"
                  style="display: block; max-width: 100%; background: none; opacity: 1; border: 0px; margin: 0px; padding: 0px;"
                />
              </span>
              <img
                alt="Imagen de Sopladora Aspiradora Philco Mjpsa116  Eléctrica 2600w 220v - 240v."
                data-nimg="intrinsic"
                decoding="async"
                src="data:image/gif;base64,R0lGODlhAQABAIAAAAAAAP///yH5BAEAAAAALAAAAAABAAEAAAIBRAA7"
                style="position: absolute; top: 0px; left: 0px; bottom: 0px; right: 0px; box-sizing: border-box; padding: 0px; margin: auto; display: block; width: 0px; height: 0px; min-width: 100%; max-width: 100%; min-height: 100%; max-height: 100%;"
              />
              <noscript />
            </span>
          </a>
        </figure>
        <div>
          <div>
            <div>
              <div>
                <div
                  aria-label="10899 pesos"
                  data-testid="priceDisplay"
                >
                  <span>
                    $
                     
                    10.899
                  </span>
                </div>
              </div>
              <div />
            </div>
            <div>
              Buenos Aires, Tristán Suarez
            </div>
          </div>
          <a
            href="/items/MLA850868389?description=Sopladora+Aspiradora+Philco+Mjpsa116+Electrica+2600w+220v+-+240v"
          >
            <h2>
              Sopladora Aspiradora Philco Mjpsa116  Eléctrica 2600w 220v - 240v
            </h2>
          </a>
        </div>
      </article>
      <article
        data-testid="listItemsItem"
      >
        <figure
          data-testid="listItemsItemImage"
        >
          <a
            href="/items/MLA1121494420?description=Sopladora+Daewoo+Daeb600+Electrica+600w+240v"
          >
            <span
              style="box-sizing: border-box; display: inline-block; overflow: hidden; background: none; opacity: 1; border: 0px; margin: 0px; padding: 0px; position: relative; max-width: 100%;"
            >
              <span
                style="box-sizing: border-box; display: block; background: none; opacity: 1; border: 0px; margin: 0px; padding: 0px; max-width: 100%;"
              >
                <img
                  alt=""
                  aria-hidden="true"
                  src="data:image/svg+xml,%3csvg%20xmlns=%27http://www.w3.org/2000/svg%27%20version=%271.1%27%20width=%27160%27%20height=%27160%27/%3e"
                  style="display: block; max-width: 100%; background: none; opacity: 1; border: 0px; margin: 0px; padding: 0px;"
                />
              </span>
              <img
                alt="Imagen de Sopladora Daewoo Daeb600  Eléctrica 600w 240v."
                data-nimg="intrinsic"
                decoding="async"
                src="data:image/gif;base64,R0lGODlhAQABAIAAAAAAAP///yH5BAEAAAAALAAAAAABAAEAAAIBRAA7"
                style="position: absolute; top: 0px; left: 0px; bottom: 0px; right: 0px; box-sizing: border-box; padding: 0px; margin: auto; display: block; width: 0px; height: 0px; min-width: 100%; max-width: 100%; min-height: 100%; max-height: 100%;"
              />
              <noscript />
            </span>
          </a>
        </figure>
        <div>
          <div>
            <div>
              <div>
                <div
                  aria-label="7820 pesos"
                  data-testid="priceDisplay"
                >
                  <span>
                    $
                     
                    7.820
                  </span>
                </div>
              </div>
              <div />
            </div>
            <div>
              Buenos Aires, San Justo
            </div>
          </div>
          <a
            href="/items/MLA1121494420?description=Sopladora+Daewoo+Daeb600+Electrica+600w+240v"
          >
            <h2>
              Sopladora Daewoo Daeb600  Eléctrica 600w 240v
            </h2>
          </a>
        </div>
      </article>
      <article
        data-testid="listItemsItem"
      >
        <figure
          data-testid="listItemsItemImage"
        >
          <a
            href="/items/MLA883353487?description=Sillas+Plasticas+Reforzadas+Apilables+Reina+Pack+10"
          >
            <span
              style="box-sizing: border-box; display: inline-block; overflow: hidden; background: none; opacity: 1; border: 0px; margin: 0px; padding: 0px; position: relative; max-width: 100%;"
            >
              <span
                style="box-sizing: border-box; display: block; background: none; opacity: 1; border: 0px; margin: 0px; padding: 0px; max-width: 100%;"
              >
                <img
                  alt=""
                  aria-hidden="true"
                  src="data:image/svg+xml,%3csvg%20xmlns=%27http://www.w3.org/2000/svg%27%20version=%271.1%27%20width=%27160%27%20height=%27160%27/%3e"
                  style="display: block; max-width: 100%; background: none; opacity: 1; border: 0px; margin: 0px; padding: 0px;"
                />
              </span>
              <img
                alt="Imagen de Sillas Plasticas Reforzadas Apilables Reina Pack 10."
                data-nimg="intrinsic"
                decoding="async"
                src="data:image/gif;base64,R0lGODlhAQABAIAAAAAAAP///yH5BAEAAAAALAAAAAABAAEAAAIBRAA7"
                style="position: absolute; top: 0px; left: 0px; bottom: 0px; right: 0px; box-sizing: border-box; padding: 0px; margin: auto; display: block; width: 0px; height: 0px; min-width: 100%; max-width: 100%; min-height: 100%; max-height: 100%;"
              />
              <noscript />
            </span>
          </a>
        </figure>
        <div>
          <div>
            <div>
              <div>
                <div
                  aria-label="20490 pesos"
                  data-testid="priceDisplay"
                >
                  <span>
                    $
                     
                    20.490
                  </span>
                </div>
              </div>
              <div />
            </div>
            <div>
              Buenos Aires, Vicente López
            </div>
          </div>
          <a
            href="/items/MLA883353487?description=Sillas+Plasticas+Reforzadas+Apilables+Reina+Pack+10"
          >
            <h2>
              Sillas Plasticas Reforzadas Apilables Reina Pack 10
            </h2>
          </a>
        </div>
      </article>
    </section>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
